src/main/java/net/minecraftforge/fml/common/Loader.java

```diff
@@ -680,78 +682,110 @@ public ModClassLoader getModClassLoader()
         return modClassLoader;
     }
 
-    public void computeDependencies(String dependencyString, Set<ArtifactVersion> requirements, List<ArtifactVersion> dependencies, List<ArtifactVersion> dependants)
+    public static void computeDependencies(String dependencyString, Set<ArtifactVersion> requirements, List<ArtifactVersion> dependencies, List<ArtifactVersion> dependants)
     {
         if (dependencyString == null || dependencyString.length() == 0)
         {
             return;
         }
 
-        boolean parseFailure = false;
-
         for (String dep : DEPENDENCYSPLITTER.split(dependencyString))
         {
             List<String> depparts = Lists.newArrayList(DEPENDENCYPARTSPLITTER.split(dep));
-            // Need two parts to the string
             if (depparts.size() != 2)
             {
-                parseFailure = true;
-                continue;
+                parseFailure(dep, "Dependecy string needs 2 parts");
             }
-            String instruction = depparts.get(0);
             String target = depparts.get(1);
             boolean targetIsAll = target.startsWith("*");
 
-            // Cannot have an "all" relationship with anything except pure *
             if (targetIsAll && target.length() > 1)
             {
-                parseFailure = true;
-                continue;
+                parseFailure(dep, "Cannot have an \"all\" relationship with anything except pure *");
+            }
+
+            if (targetIsAll && target.indexOf('@') > -1)
+            {
+                parseFailure(dep, "You cannot have a versioned dependency on everything");
             }
 
-            // If this is a required element, add it to the required list
-            if ("required-before".equals(instruction) || "required-after".equals(instruction))
+            List<String> instructions = Lists.newArrayList(DEPENDENCYINSTRUCTIONSSPLITTER.split(depparts.get(0)));
+            if (!DEPENDENCYINSTRUCTIONS.containsAll(instructions))
             {
-                // You can't require everything
-                if (!targetIsAll)
+                parseFailure(dep, String.format("Found invalid instructions. Only %s are allowed.", DEPENDENCYINSTRUCTIONS.toString().replace('[', ' ').replace(']', ' ').trim()));
+            }
+            boolean hasSide = false;
+            boolean hasRequired = false;
+            boolean hasOrder = false;
+            for (String instruction : instructions)
```

Comment:
No, I mean in `Spliterator` from Java 8, there is an integer operated by binary operations. Each bit of that integer represents a feature of the `Spliterator`. Likewise, you can first search for all instructions, and make the code cleaner (The point is to make the code clean)